### PR TITLE
chore: fix audit high and critical issues

### PR DESCRIPTION
**Pull request by ZedMattr**
## Description

Need to update jest / ts-jest to 28+ to solve some security issues caused by them, means we would have to drop node support 11 and so on.

Dropped dev-dep of cz-conventional-changelog.

Lodash is a common one.

After changes audit report:
image

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] Changes follow the **[contributing](../docs/CONTRIBUTING.md)** document.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Which merge strategy will you use?

- [ ] Squash
- [x] Rebase (REVIEW COMMITS)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mattrglobal/bbs-signatures",
-  "version": "1.0.0",
```
Comment (kenhuang) [inline package.json:3]: Should this upgrade count as breaking changes?

Comment (ZedMattr) [reply package.json:3]: Was checking previously with others, so likely will be considered as breaking changes as we will be dropping support for old node versions.

But will double confirm again with @jhill-mattr @tplooker @colinyip 

Thanks for checking & raising this @kenhuang

Comment (jhill-mattr) [reply package.json:3]: Could be argued both ways. I have checked with @tplooker and we agree no downside to bumping major version.

File: package.json [hunk 2]
```diff
@@ -36,34 +36,34 @@
...
-    "hooks": {
-      "commit-msg": "[[ -n $HUSKY_BYPASS ]] || commitlint -E HUSKY_GIT_PARAMS",
-      "pre-commit": "[[ -n $HUSKY_BYPASS ]] || pretty-quick --staged"
-    }
+  "resolutions": {
+    "cz-conventional-changelog/**/minimist": "1.2.6",
+    "cz-conventional-changelog/**/merge": "2.1.1",
+    "lodash": "4.17.21"
   },
   "commitlint": {
```
Comment (jhill-mattr) [inline package.json:68]: Is this still needed now you have added `commitlint.config.js`

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mattrglobal/bbs-signatures",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (jhill-mattr) [inline package.json:3]: This needs to happen using the release process, not in this PR

File: package.json (hunk 2 above)
Comment (jhill-mattr) [reply package.json:68]: Bump

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+[[ -n $HUSKY_BYPASS ]] || npx pretty-quick --staged
```
Comment (pablodenadai) [inline .husky/pre-commit:4]: > 

> 
```suggestion
[[ -n $HUSKY_BYPASS ]] || npx --no -- pretty-quick --staged
```

File: .husky/commit-msg [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+[[ -n $HUSKY_BYPASS ]] || npx commitlint --edit $1
```
Comment (pablodenadai) [inline .husky/commit-msg:4]:
```suggestion
[[ -n $HUSKY_BYPASS ]] || npx --no -- commitlint --edit $1
```

File: package.json [hunk 6]
```diff
@@ -36,39 +36,31 @@
...
-    }
-  },
-  "commitlint": {
-    "extends": [
-      "@commitlint/config-conventional"
-    ]
+  "resolutions": {
+    "lodash": "4.17.21"
   },
   "config": {
```
Comment (pablodenadai) [inline package.json:65]: This can be removed now that we don't have `cz-conventional-changelog`

**Review by pablodenadai — APPROVED** [review]

**Pull request merged**